### ISSUE TITLE
transformPosition edge case ?

### ISSUE THREAD

**Issue by antoinelyset** [ottypes/json1#27]
Hi Joseph! 

As always, thanks for this library. I'm having a bit of trouble knowing if I encountered a bug or I just use the library badly. What do you think about this example?

```javascript
const doc = {
  object: 'one',
  nodes: [
    {
      object: 'two',
      nodes: [
        {
          object: 'three',
          nodes: [
            {
              object: 'four',
              nodes: [{ object: 'five' }],
            },
          ],
        },
      ],
    },
  ],
}

const op = [
  'nodes',
  0,
  { r: true, d: 0 },
  'nodes',
  [0, { p: 0, i: { object: 'six' } }],
  [2, { i: { object: 'seven' } }],
]

const appliedDoc = {
  "object": "one",
  "nodes": [
    {
      "object": "three",
      "nodes": [
        {
          "object": "six"
        },
        {
          "object": "four",
          "nodes": [
            {
              "object": "five"
            }
          ]
        },
        {
          "object": "seven"
        }
      ]
    }
  ]
}

const path = ['nodes', 0, 'nodes' ,0, 'nodes', 0, 'nodes', 0]

json1.type.transformPosition(path, op)
// => [ 'nodes', 0, 'nodes', 0, 'nodes', 1 ] which seems buggy
```

Thanks for your time.

**Comment by josephg** [comment]
Hey sorry I didn’t get back to this. That example looks really complicated and hard to follow. Can you simplify it to a minimal case where transformPosition does something unexpected?

**Comment by antoinelyset** [comment]
Hey. Sorry for making this over complicated indeed.

I think I had it simplified to this (it was extracted from a "real-world" case) : 

```javascript
const doc = {
  object: "one",
  nodes: [
    {
      object: "two",
      nodes: [
        {
          object: "three"
        }
      ]
    }
  ]
}

const op = [
  { r: true, d: 0 },
  "nodes",
  0,
  { p: 0, i: { object: "four" } }
]

const appliedDoc = {
  object: "two",
  nodes: [
    {
      object: "four"
    },
    {
      object: "three"
    }
  ]
}

const path = ["nodes", 0, "nodes", 0]
json1.type.transformPosition(path, op)
// => ["nodes", 0]
// I think it should be ["nodes", 1]
```

The remove and pick are complicated since `nodes` array is not the same at pickup time as at drop time.

**Comment by josephg** [comment]
Yes that does look like a bug!

**Comment by josephg** [comment]
Fixed and published 📦 ot-json1@1.0.2

Hopefully I didn't break anything in the process! It looks like the test suite for transformPosition isn't as detailed as I'd like.

**Comment by antoinelyset** [comment]
Thanks a lot for your work. 🙏
